[BugFix] Fix incorrect slot id for auto increment column in plan when loading

srlch opened the pull request:
## Why I'm doing:
In current impl, we use slot offset/index id in tupleDescriptor as the slot id for auto increment column loading plan. But it is not correct and maybe get exception when loading.

## What I'm doing:
Use correct slot id.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0

banmoy commented:
When does this bug occur?  It seems that this issue has existed for a long time.

srlch commented:
@banmoy yes, it has existed for a long time. If this bug was triggered, non-nullable column _validate_data will fail, `NULL value in non-nullable column` will be throw.

wyb commented:
add UT?

srlch committed:
[BugFix] Fix incorrect slot id for auto increment column in plan for loading

## Why I'm doing:
In current impl, we use slot offset/index id in tupleDescriptor as the slot id for auto increment column
loading plan. But it is not correct and maybe get exception when loading.

## What I'm doing:
Use correct slot id.

Signed-off-by: srlch <linzichao@starrocks.com>
